drivers/gpio/gpio_pca_series.c

```diff
@@ -0,0 +1,2204 @@
+	gpio_pca_series_reg_write(dev, PCA_REG_TYPE_2B_OUTPUT_DRIVE_STRENGTH, reset_value_1);
+	gpio_pca_series_reg_write(dev, PCA_REG_TYPE_1B_INPUT_LATCH, reset_value_0);
+	gpio_pca_series_reg_write(dev, PCA_REG_TYPE_1B_PULL_ENABLE, reset_value_0);
+	gpio_pca_series_reg_write(dev, PCA_REG_TYPE_1B_PULL_SELECT, reset_value_1);
+	gpio_pca_series_reg_write(dev, PCA_REG_TYPE_1B_OUTPUT_CONFIG, reset_value_0);
+#ifdef CONFIG_GPIO_PCA_SERIES_INTERRUPT
+	gpio_pca_series_reg_write(dev, PCA_REG_TYPE_1B_INTERRUPT_MASK, reset_value_1);
+	gpio_pca_series_reg_write(dev, PCA_REG_TYPE_1B_INTERRUPT_STATUS, reset_value_0);
+	gpio_pca_series_reg_write(dev, PCA_REG_TYPE_2B_INTERRUPT_EDGE, reset_value_1);
+#endif /* CONFIG_GPIO_PCA_SERIES_INTERRUPT */
+}
+
+void gpio_pca_series_dump(const struct device *dev)
+{
+	const struct gpio_pca_series_config *cfg = dev->config;
+	int ret;
+
+	LOG_INF("**** device debug dump ****");
+	LOG_INF("part_no: %s", gpio_pca_series_part_name[cfg->part_cfg->port_no]);
+	LOG_INF("register profile:");
+	LOG_INF("type\t"
+		"name\t"
+		"address\t"
+#ifdef CONFIG_GPIO_PCA_SERIES_CACHE_ALL
+		"cache\t"
+#endif /* CONFIG_GPIO_PCA_SERIES_CACHE_ALL */
+	);
+	for (int reg_type = 0; reg_type < PCA_REG_TYPE_COUNT; ++reg_type) {
+		uint8_t reg = cfg->part_cfg->regs[reg_type];
+		uint8_t reg_size = gpio_pca_series_reg_get_size(dev, reg_type);
+		uint8_t reg_val[8];
+#ifdef CONFIG_GPIO_PCA_SERIES_CACHE_ALL
+		uint8_t cache;
+		uint8_t cache_val[8];
+
+		ret = gpio_pca_series_reg_cache_get_offset(dev, reg_type, &cache);
+		if (ret != 0) {
+			cache = PCA_REG_INVALID;
+		}
+		if (reg == PCA_REG_INVALID && cache == PCA_REG_INVALID)
+#else
+		if (reg == PCA_REG_INVALID)
+#endif /* CONFIG_GPIO_PCA_SERIES_CACHE_ALL */
+		{
+			continue;
+		}
+
+		if (reg != PCA_REG_INVALID) {
+			ret = gpio_pca_series_reg_read(dev, reg_type, reg_val);
+			if (ret != 0) {
+				LOG_ERR("read reg error from reg type %d, invalidate this reg",
+					reg_type);
+				reg = PCA_REG_INVALID;
+			}
+		}
+#ifdef CONFIG_GPIO_PCA_SERIES_CACHE_ALL
+		if (cache != PCA_REG_INVALID) {
+			ret = gpio_pca_series_reg_read_cache(dev, reg_type, cache_val);
+			if (ret != 0) {
+				LOG_ERR("read reg cache error from reg type %d, invalidate this "
+					"reg cache",
+					reg_type);
+				reg = PCA_REG_INVALID;
+			}
+		}
+#endif /* CONFIG_GPIO_PCA_SERIES_CACHE_ALL */
+
+		/** do_print */
+		LOG_INF("%.2d\t"
+			"%s\t"
+			"0x%2.2x\t"
+#ifdef CONFIG_GPIO_PCA_SERIES_CACHE_ALL
+			"0x%2.2x\t"
+#endif /* CONFIG_GPIO_PCA_SERIES_CACHE_ALL */
+			,
+			reg_type, gpio_pca_series_reg_name[reg_type], reg
+#ifdef CONFIG_GPIO_PCA_SERIES_CACHE_ALL
+			,
+			cache
+#endif /* CONFIG_GPIO_PCA_SERIES_CACHE_ALL */
+		);
+	}
+}
+
+/**
+ * @brief configure output drive strength
+ *
+ * @param dev
+ * @param pin
+ * @param strength
+ * @return int 0 if success
+ *             -EIO if i2c error
+ */
+static int gpio_pca_series_pin_config_drive_strength(
```

Comment:
This one is not used as well, as zephyr does not provide api for that. Not sure if it should be kept.

Reply:
No, we should drop code that will not be used by Zephyr

Reply:
Setting drive strength is common feature to gpio controllers. For SoC buit-in gpio controllers, this is usually set by pinctrl driver. But for gpio expander this does not apply. I went through issues and found #17072, which makes it possible to expand driver-specific api. Is it possible to expand gpio api with drive strength like that?

Reply:
sure, let's leave it to another pr.

Reply:
removed in this pr.